packages/runner/src/traverse.ts

```diff
@@ -625,6 +626,52 @@ export class SchemaObjectTraverser<K, S> extends BaseObjectTraverser<K, S> {
     }
   }
 
+  static resolveSchemaRef(
+    schemaContext: Readonly<SchemaContext>,
+  ): Readonly<SchemaContext> | undefined {
+    if (
+      !isObject(schemaContext.schema) || !("$ref" in schemaContext.schema) ||
+      schemaContext.schema["$ref"] === undefined
+    ) {
+      // We shouldn't have called this function, but just return it as-is
+      return schemaContext;
+    }
+    if (schemaContext.rootSchema === undefined) {
+      logger.warn(() => ["Unsupported $ref without root schema: ", schemaRef]);
+      return undefined;
+    }
+    const schemaRef = schemaContext.schema["$ref"];
+    if (schemaRef === "#") {
+      return {
+        schema: schemaContext.rootSchema,
+        rootSchema: schemaContext.rootSchema,
+      };
+    } else if (
+      schemaRef.startsWith("#/definitions/") || schemaRef.startsWith("#/$defs/")
```

Comment:
same, let's leave out $defs as you suggested?

more generally i wonder whether this should be a condition though, we could already say anything starting with `#/` is fine?